service/history/workflow/mutable_state_impl.go

```diff
@@ -5988,13 +6026,11 @@ func (ms *MutableStateImpl) closeTransaction(
 		return closeTransactionResult{}, err
 	}
 
-	if ms.isStateDirty() {
-		if err := ms.closeTransactionUpdateTransitionHistory(
-			transactionPolicy,
-		); err != nil {
-			return closeTransactionResult{}, err
-		}
-		ms.closeTransactionHandleUnknownVersionedTransition()
+	isStateDirty := ms.isStateDirty()
```

Comment:
move to L6041?

Reply:
no, it needs to be here.  `closeTransactionPrepareEvents()` and `chasmTree.CloseTransaction()` changes the state of history builder and chasm tree and make them "clean"

I will add a comment here.